extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskSqlEngine.java

```diff
@@ -220,13 +224,18 @@ private static void validateInsert(
       );
     }
     catch (Exception e) {
-      throw new ValidationException(
-          StringUtils.format(
-              "Invalid segmentGranularity: %s",
-              plannerContext.queryContext().get(DruidSqlInsert.SQL_INSERT_SEGMENT_GRANULARITY)
-          ),
-          e
-      );
+      // This is a defensive check as the DruidSqlInsert.SQL_INSERT_SEGMENT_GRANULARITY in the query context is
+      // populated by Druid. If the user entered an incorrect granularity, that should have been flagged before reaching
+      // here
```

Comment:
You can put all of this into the message instead of a comment.  That way, if this ever does get fired, the person who sees it will know without needing to find this comment in the code.  Also, if it's in the message, you don't need a comment in the code.


For example, you could say "Unable to parse the segmentGranularity [%s].  This should have been created by class XYZ, so this exception most likely indicates a bug over there."

Given that this is a developer-facing message, it's perfectly fine to include such details about the code and expectations of the code.